src/pages/Results/pages/Results/pages/FootO/pages/Splits/components/FootOSplitsTable/FootOSplitsTable.tsx

```diff
@@ -64,91 +61,344 @@ export default function FootOSplitsTable(props: FootOSplitsTableProps) {
+        const { clientX } = e
+        // Next, figure out the distance between the top of the track and the top of the viewport
+        const target = e.target as HTMLDivElement
+        const rect = target.getBoundingClientRect()
+        const trackLeft = rect.left
+        // We want the middle of the thumb to jump to where we clicked, so we subtract half the thumb's height to offset the position
+        const thumbOffset = -(thumbWidth / 2)
+        // Find the ratio of the new position to the total content length using the thumb and track values...
+        const clickRatio = (clientX - trackLeft + thumbOffset) / trackCurrent.clientWidth
+        // ...so that you can compute where the content should scroll to.
+        const scrollAmount = Math.floor(clickRatio * contentCurrent.scrollWidth)
+        // And finally, scroll to the new position!
+        contentCurrent.scrollTo({
+          left: scrollAmount,
+          behavior: "smooth",
+        })
+        headerCurrent.scrollTo({
+          left: scrollAmount,
+          behavior: "smooth",
+        })
+      }
+    },
+    [thumbWidth],
+  )
+
+  const handleThumbMousedown = useCallback((e: React.MouseEvent<HTMLDivElement>) => {
+    e.preventDefault()
+    e.stopPropagation()
+    setScrollStartPosition(e.clientX)
+    if (bodyRef.current) setInitialScrollLeft(bodyRef.current.scrollLeft)
+    if (headerRef.current) setInitialScrollLeft(headerRef.current.scrollLeft)
+    setIsDragging(true)
+  }, [])
+
+  const handleThumbMouseup = useCallback(
+    (e: MouseEvent) => {
+      e.preventDefault()
+      e.stopPropagation()
+      if (isDragging) {
+        setIsDragging(false)
+      }
+    },
+    [isDragging],
+  )
+
+  const handleThumbMousemove = useCallback(
+    (e: MouseEvent) => {
+      e.preventDefault()
+      e.stopPropagation()
+      if (isDragging) {
+        const { scrollWidth: contentScrollWidth, offsetWidth: contentOffsetWidth } =
+          bodyRef.current!
+
+        const deltaX = (e.clientX - scrollStartPosition!) * (contentOffsetWidth / thumbWidth)
+        const newScrollLeft = Math.min(
+          initialScrollLeft + deltaX,
+          contentScrollWidth - contentOffsetWidth,
+        )
+
+        bodyRef.current!.scrollLeft = newScrollLeft
+        headerRef.current!.scrollLeft = newScrollLeft
+      }
+    },
+    [isDragging, scrollStartPosition, thumbWidth, initialScrollLeft],
+  )
+
+  // Listen for mouse events to handle scrolling by dragging the thumb
+  useEffect(() => {
+    document.addEventListener("mousemove", handleThumbMousemove)
+    document.addEventListener("mouseup", handleThumbMouseup)
+    document.addEventListener("mouseleave", handleThumbMouseup)
+    return () => {
+      document.removeEventListener("mousemove", handleThumbMousemove)
+      document.removeEventListener("mouseup", handleThumbMouseup)
+      document.removeEventListener("mouseleave", handleThumbMouseup)
+    }
+  }, [handleThumbMousemove, handleThumbMouseup])
```

Comment:
This code handles the ScrollBar, should be somewhere else, for sure, but I still dunno much about react and If I can extract it out of here